### ISSUE TITLE
Github-flavored markdown support

### ISSUE THREAD

**Issue by Yamakaky** [onur/docs.rs#53]
Example: `[x]` in https://docs.rs/crate/dcpu/0.1.1

**Comment by onur** [comment]
docs.rs is currently using hoedown markdown renderer. It is same renderer used in rustdoc.

Looks like CommonMark support is under development: hoedown/hoedown#119. But hoedown development seems stalled for a while. If rust gets a better markdown renderer, I'll switch to it.

**Comment by Yamakaky** [comment]
ok

**Comment by kivikakk** [comment]
Just as an FYI, there's now a fully CommonMark compatible (and [GFM compatible](https://github.com/blog/2333-a-formal-spec-for-github-flavored-markdown)) parser/renderer written in Rust: https://github.com/kivikakk/comrak.

It's a straight Rust port of [`cmark-gfm`](https://github.com/github/cmark), which itself is a fork of the reference C implementation of CommonMark. I still have a lot of work to do in terms of performance, but it's 100% spec compliant.

**Comment by onur** [comment]
@kivikakk comrak seems promising. I hope you can add library interface.

**Comment by kivikakk** [comment]
@onur Happy to add a library interface! I'll try to get something out onto crates.io soon.

**Comment by onur** [comment]
@kivikakk I tried comrak and I think only missing feature is tasklist extension.

**Comment by kivikakk** [comment]
Peeking at `src/web/mod.rs`, I see here's the currently used extensions:

``` rust
        let mut extensions = Extension::empty();
        extensions.insert(FENCED_CODE);
        extensions.insert(FOOTNOTES);
        extensions.insert(SUPERSCRIPT);
        extensions.insert(TABLES);
        extensions.insert(AUTOLINK);
        extensions.insert(NO_INTRA_EMPHASIS);
```

Missing would be footnotes, superscript, and no intra-emphasis. I'd be able to add these pretty easily; task lists too, if there's a reference implementation somewhere.

**Comment by kivikakk** [comment]
Tasklists are now in master.

**Comment by kivikakk** [comment]
I've added superscript as well (left out footnotes for now as they're huge, and no-intra-emphasis isn't strictly necessary in CommonMark), and released 0.1.2 to crates.io.
